src/mlpack/methods/random_forest/random_forest_regressor.hpp

```diff
@@ -0,0 +1,387 @@
+/**
+ * @file methods/random_forest/random_forest_regressor.hpp
+ * @author Dinesh Kumar
+ *
+ * Definition of the RandomForestRegressor class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_RANDOM_FOREST_RANDOM_FOREST_REGRESSOR_HPP
+#define MLPACK_METHODS_RANDOM_FOREST_RANDOM_FOREST_REGRESSOR_HPP
+
+#include <mlpack/methods/decision_tree/decision_tree_regressor.hpp>
+#include "bootstrap.hpp"
+
+namespace mlpack {
+
+/**
+ * This class implements a random forest regressor
```

Comment:
Can you add some more details here? You can look at https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/linear_regression/linear_regression.hpp#L25 as a reference.

Reply:
Would be great to have a description for the none trivial template parameters.